sources/model/managars/GGHTTPClientManager.m

```diff
@@ -568,7 +569,25 @@ - (void)sendFindMeRequestWithFindMeConfiguration:(nonnull GGFindMe *)findmeConfi
                    //
                }];
 }
+-(void)sendMaskedNumberRequestForOrderWithUUID:(NSString *_Nonnull)uuid
+                                forPhoneNumber:(NSString*_Nonnull)originalPhoneNumber
+                         withCompletionHandler:(nullable GGMaskedPhoneNumberResponseHandler)completionHandler{
 
+    NSMutableDictionary *params = [NSMutableDictionary dictionary];
+    [self addAuthinticationToParams:&params];
+    [self httpRequestWithMethod:BCRESTMethodGet
+                           path:[NSString stringWithFormat:API_PATH_SHARED_MASK_PHONE, uuid]
+                         params:params
+              completionHandler:^(BOOL success, id JSON, NSError *error) {
+                  
+                  // update last date
+                  self.lastEventDate = [NSDate date];
+                  NSString* phoneNumber = [JSON objectForKey:@"phone_number"];
```

Comment:
i would use the JSON util here to save crashes when that value will not be a String